bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/OpenThermGatewaySocketConnector.java

```diff
@@ -61,7 +63,56 @@
     private @Nullable Future<Boolean> future;
     private @Nullable ExecutorService executor;
 
-    private Map<String, Entry<Long, GatewayCommand>> pendingCommands = new ConcurrentHashMap<>();
+    /**
+     * FIFO queue of commands that are pending being sent to the gateway. That is commands that are either not yet sent,
+     * or sent but not yet acknowledged and pending possible re-sending.
+     *
+     * Note: we must use 'synchronized' when accessing this object to ensure proper thread safety.
+     */
+    private final List<PendingCommand> pendingCommands = new ArrayList<>();
+
+    /**
+     * Wrapper for a command entry in the pending command FIFO queue.
+     *
+     * @author AndrewFG - initial contribution
+     */
+    private class PendingCommand {
+        protected final GatewayCommand command;
+        protected final long expiryTime = System.currentTimeMillis() + COMMAND_RESPONSE_MAX_WAIT_TIME_MILLISECONDS;
```

Comment:
Consider if `Instant` would be more reliable (and perhaps even readable) as based on a timeline, not wall clock: https://www.baeldung.com/java-measure-elapsed-time. See also #12493.